### PR TITLE
[FR] rust end of chapter 8

### PR DESCRIPTION
**Pull request by mgemard**
Finished chapter 8 of Tour of Rust

### CODE REVIEW

File: lessons.yaml [hunk 1]
```diff
@@ -6263,6 +6263,26 @@ pages:
...
         
+    fr:
+      title: Partage entre Fil d'Exécution (Thread)
+      code: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use%20std%3A%3Async%3A%3AMutex%3B%0A%0Astruct%20Pie%3B%0A%0Aimpl%20Pie%20%7B%0A%20%20%20%20fn%20eat(%26self)%20%7B%0A%20%20%20%20%20%20%20%20println!(%22Je%20suis%20le%20seul%20%C3%A0%20pouvoir%20manger%20la%20tarte!%22)%3B%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20mutex_pie%20%3D%20Mutex%3A%3Anew(Pie)%3B%0A%20%20%20%20%0A%20%20%20%20%2F%2F%20On%20emprunte%20une%20r%C3%A9f%C3%A9rence%20immuable.%20Celle-ci%20doit%20%C3%AAtre%0A%20%20%20%20%2F%2F%20v%C3%A9rouill%C3%A9e%20(en%20anglais%20locked).%20On%20doit%20%C3%A9galement%20faire%0A%20%20%20%20%2F%2F%20appel%20%C3%A0%20%27unwrap%27%20car%20l%27appel%20%C3%A0%20%27lock%27%20peut%20%C3%A9chouer.%0A%20%20%20%20let%20ref_pie%20%3D%20mutex_pie.lock().unwrap()%3B%0A%20%20%20%20%0A%20%20%20%20ref_pie.eat()%3B%0A%20%20%20%20%0A%20%20%20%20%2F%2F%20La%20r%C3%A9f%C3%A9rence%20ref_pie%20est%20d%C3%A9sallou%C3%A9e%20ici.%20mutex_pie%0A%20%20%20%20%2F%2F%20pourrait%20de%20nouveau%20%C3%AAtre%20utilis%C3%A9%20ici.%0A%20%20%20%20%0A%7D%0A
```
Comment (VincentFoulon80) [inline lessons.yaml:6268]:
```suggestion
      code: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use%20std%3A%3Async%3A%3AMutex%3B%0A%0Astruct%20Pie%3B%0A%0Aimpl%20Pie%20%7B%0A%20%20%20%20fn%20eat(%26self)%20%7B%0A%20%20%20%20%20%20%20%20println!(%22Je%20suis%20le%20seul%20%C3%A0%20pouvoir%20manger%20la%20tarte!%22)%3B%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20mutex_pie%20%3D%20Mutex%3A%3Anew(Pie)%3B%0A%20%20%20%20%0A%20%20%20%20%2F%2F%20On%20emprunte%20une%20r%C3%A9f%C3%A9rence%20immuable.%20Celle-ci%20doit%20%C3%AAtre%0A%20%20%20%20%2F%2F%20verrouill%C3%A9e%20(en%20anglais%20locked).%20On%20doit%20%C3%A9galement%20faire%0A%20%20%20%20%2F%2F%20appel%20%C3%A0%20'unwrap'%20car%20l'appel%20%C3%A0%20'lock'%20peut%20%C3%A9chouer.%0A%20%20%20%20let%20ref_pie%20%3D%20mutex_pie.lock().unwrap()%3B%0A%20%20%20%20%0A%20%20%20%20ref_pie.eat()%3B%0A%20%20%20%20%0A%20%20%20%20%2F%2F%20La%20r%C3%A9f%C3%A9rence%20ref_pie%20est%20d%C3%A9sallou%C3%A9e%20ici.%20mutex_pie%0A%20%20%20%20%2F%2F%20pourrait%20de%20nouveau%20%C3%AAtre%20utilis%C3%A9%20ici.%0A%20%20%20%20%0A%7D%0A
```

File: lessons.yaml [hunk 2]
```diff
@@ -6263,6 +6263,26 @@ pages:
...
+      content_markdown: |
+        `Mutex` est une structure de données (couramment détenue par des pointeurs
+        intelligents) pouvant contenir des données et nous permet d'emprunter des
+        références mutable ou immuables vers les données qu'elle contient. `Mutex`
+        restreint l'accès aux données à un seul thread CPU, bloquant les autres
+        threads jusqu'à ce que le thread d'origine ait terminé et déverrouillé les
+        données.
+
+        Le multithreading dépasse le cadre de Tour of Rust, mais `Mutex` est un
+        élément fondamental de la programmation concurrentiels avec Rust.
```
Comment (VincentFoulon80) [inline lessons.yaml:6278]:
```suggestion
        Le multithreading dépasse le cadre de Tour of Rust, mais `Mutex` est un
        élément fondamental de la programmation concurrentielle avec Rust.
```

File: lessons.yaml [hunk 3]
```diff
@@ -6291,6 +6311,34 @@ pages:
...
...0pour%0A%20%20%20%20%20%20%20%20%2F%2F%20effectuer%20un%20emprunt%20mutable.%0A%20%20%20%20%20%20%20%20let%20mut%20p%20%3D%20self.pie.borrow_mut()%3B%0A%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20%2F%2F%20On%20modifie%20la%20donn%C3%A9e.%0A%20%20%20%20%20%20%20%20p.eat_slice(%26self.name)%3B%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20pie%20%3D%20Rc%3A%3Anew(RefCell%3A%3Anew(Pie%20%7B%20slices%3A%208%20%7D))%3B%0A%20%20%20%20%0A%20%20%20%20%2F%2F%20Ferris%20et%20Sara%20sont%20cr%C3%A9es%20avec%20des%20clones%20de%0A%20%20%20%20%2F%2F%20pointeurs%20intelligents%20vers%20Pie.%0A%20%20%20%20let%20ferris%20%3D%20SeaCreature%20%7B%0A%20%20%20%20%20%20%20%20name%3A%20String%3A%3Afrom(%22Ferris%22)%2C%0A%20%20%20%20%20%20%20%20pie%3A%20pie.clone()%2C%0A%20%20%20%20%7D%3B%0A%20%20%20%20let%20sara%20%3D%20SeaCreature%20%7B%0A%20%20%20%20%20%20%20%20name%3A%20String%3A%3Afrom(%22Sara%22)%2C%0A%20%20%20%20%20%20%20%20pie%3A%20pie.clone()%2C%0A%20%20%20%20%7D%3B%0A%20%20%20%20%0A%20%20%20%20ferris.eat()%3B%0A%20%20%20%20sara.eat()%3B%0A%0A%20%20%20%20let%20p%20%3D%20pie.borrow()%3B%0A%20%20%20%20println!(%22%7B%7D%20parts%20restantes.%22%2C%20p.slices)%3B%0A%7D%0A
```
Comment (VincentFoulon80) [inline lessons.yaml:6316]:
```suggestion
      code: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use%20std%3A%3Acell%3A%3ARefCell%3B%0Ause%20std%3A%3Arc%3A%3ARc%3B%0A%0Astruct%20Pie%20%7B%0A%20%20%20%20slices%3A%20u8%2C%0A%7D%0A%0Aimpl%20Pie%20%7B%0A%20%20%20%20fn%20eat_slice(%26mut%20self%2C%20name%3A%20%26str)%20%7B%0A%20%20%20%20%20%20%20%20println!(%22%7B%7D%20a%20pris%20une%20part!%22%2C%20name)%3B%0A%20%20%20%20%20%20%20%20self.slices%20-%3D%201%3B%0A%20%20%20%20%7D%0A%7D%0A%0Astruct%20SeaCreature%20%7B%0A%20%20%20%20name%3A%20String%2C%0A%20%20%20%20pie%3A%20Rc%3CRefCell%3CPie%3E%3E%2C%0A%7D%0A%0Aimpl%20SeaCreature%20%7B%0A%20%20%20%20fn%20eat(%26self)%20%7B%0A%20%20%20%20%20%20%20%20%2F%2F%20Utilise%20un%20pointeur%20intelligent%20vers%20Pie%20pour%0A%20%20%20%20%20%20%20%20%2F%2F%20effectuer%20un%20emprunt%20mutable.%0A%20%20%20%20%20%20%20%20let%20mut%20p%20%3D%20self.pie.borrow_mut()%3B%0A%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20%2F%2F%20On%20modifie%20la%20donn%C3%A9e.%0A%20%20%20%20%20%20%20%20p.eat_slice(%26self.name)%3B%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20pie%20%3D%20Rc%3A%3Anew(RefCell%3A%3Anew(Pie%20%7B%20slices%3A%208%20%7D))%3B%0A%20%20%20%20%0A%20%20%20%20%2F%2F%20Ferris%20et%20Sara%20sont%20cr%C3%A9%C3%A9s%20avec%20des%20clones%20de%0A%20%20%20%20%2F%2F%20pointeurs%20intelligents%20vers%20Pie.%0A%20%20%20%20let%20ferris%20%3D%20SeaCreature%20%7B%0A%20%20%20%20%20%20%20%20name%3A%20String%3A%3Afrom(%22Ferris%22)%2C%0A%20%20%20%20%20%20%20%20pie%3A%20pie.clone()%2C%0A%20%20%20%20%7D%3B%0A%20%20%20%20let%20sara%20%3D%20SeaCreature%20%7B%0A%20%20%20%20%20%20%20%20name%3A%20String%3A%3Afrom(%22Sara%22)%2C%0A%20%20%20%20%20%20%20%20pie%3A%20pie.clone()%2C%0A%20%20%20%20%7D%3B%0A%20%20%20%20%0A%20%20%20%20ferris.eat()%3B%0A%20%20%20%20sara.eat()%3B%0A%0A%20%20%20%20let%20p%20%3D%20pie.borrow()%3B%0A%20%20%20%20println!(%22%7B%7D%20parts%20restantes.%22%2C%20p.slices)%3B%0A%7D%0A
```

File: lessons.yaml [hunk 4]
```diff
@@ -6291,6 +6311,34 @@ pages:
...
+      content_markdown: |
+        Les pointeurs intelligents peuvent sembler limités, mais ils peuvent
+        utilisé de manière très puissantes avec les structures que nous venons
+        de voir.
```
Comment (VincentFoulon80) [inline lessons.yaml:6320]:
```suggestion
        Les pointeurs intelligents peuvent sembler limités, mais ils peuvent
        être utilisés de manière très puissante avec les structures que nous venons
        de voir.
```

File: lessons.yaml [hunk 5]
```diff
@@ -6291,6 +6311,34 @@ pages:
...
+        emprunter le même vecteur de données immuables. Pour rappel `Rc` place les
+        données sur le tas.
+
+        `Rc<RefCell<Foo>>` - Permet à plusieurs pointeurs intelligents d'emprunter
+        mutuellement ou immuablement la même structure `Foo`.
+
+        `Arc<Mutex<Foo>>` - Permet à un pointeur intelligent de verrouiller
+        temporairement une données (celle-ci ne pourra plus être emprunté mutablement
+        ou immuablement) de manière exclusive à un thread du CPU (les autres threads
+        n'y aurons pas accès).
```
Comment (VincentFoulon80) [inline lessons.yaml:6332]:
```suggestion
        `Arc<Mutex<Foo>>` - Permet à un pointeur intelligent de verrouiller
        temporairement une donnée (celle-ci ne pourra plus être emprunté mutablement
        ou immuablement) de manière exclusive à un thread du CPU (les autres threads
        n'y auront pas accès).
```

File: lessons.yaml [hunk 6]
```diff
@@ -6291,6 +6311,34 @@ pages:
...
+
+        Détail de la mémoire:
+        * Vous remarquerez un thème commun avec plusieurs de ces utilisations:
+          l'utilisation d'un type de données immuable (possiblement emprunté par
+          plusieurs pointeurs intelligents) pour modifier les données internes.
+          C'est ce qu'on appel la "mutabilité intérieure" (en anglais interior
+          mutability) de Rust. C'est un modèle qui nous permet de contourner les
+          règles d'utilisation de la mémoire à l'exécution mais en gardant le même
+          niveau de sécurité que Rust nous offre lors de la compilation.
```
Comment (VincentFoulon80) [inline lessons.yaml:6341]:
```suggestion
        * Vous remarquerez un thème commun avec plusieurs de ces utilisations:
          l'utilisation d'un type de données immuable (possiblement emprunté par
          plusieurs pointeurs intelligents) pour modifier les données internes.
          C'est ce qu'on appelle la "mutabilité intérieure" (en anglais interior
          mutability) de Rust. C'est un modèle qui nous permet de contourner les
          règles d'utilisation de la mémoire à l'exécution mais en gardant le même
          niveau de sécurité que Rust nous offre lors de la compilation.
```

File: lessons.yaml [hunk 7]
```diff
@@ -6312,6 +6360,15 @@ pages:
         larger projects. In chapter 9 we break free of single page
         lines of code.
         
+    fr:
+      title: Chapitre 8 Conclusion
```
Comment (VincentFoulon80) [inline lessons.yaml:6364]:
```suggestion
      title: Chapitre 8 - Conclusion
```

**Comment by mgemard** [comment]
I has been merge too soon...

I will make a new merge request with the changes of Vincent.

**Pull request merged**